### PR TITLE
[eas-cli] disable eas build & submit for non-paying users

### PR DESCRIPTION
**Pull request by dsokal**
For Open Preview, we want to allow using EAS services (Build and Submit) only for paid (_paying?_) users.

- https://www.notion.so/expo/Allowlist-Free-Priority-Plan-for-existing-external-testers-for-EAS-Build-5b69631944134d8bb71a0112ef316dfc
- https://www.notion.so/expo/Gate-Open-Preview-behind-Dev-Services-558c8aefb4f348d9b8ab3739a867810c

API changes are in https://github.com/expo/universe/pull/6477

### CODE REVIEW

File: packages/eas-cli/src/commands/build/create.ts
Comment (fson) [inline packages/eas-cli/src/commands/build/create.ts:49]: Maybe a link here where you can "enroll in EAS"?

File: packages/eas-cli/src/commands/submit.ts
Comment (fson) [inline packages/eas-cli/src/commands/submit.ts:171]: A link here as well, maybe?

File: packages/eas-cli/src/project/isEasEnabledForProject.ts
Comment (fson) [inline packages/eas-cli/src/project/isEasEnabledForProject.ts:12]: I commented in the server side PR – would be good to get this information as a part of the GraphQL query we make in `getProjectIdAsync`. I know this is temporary, but since we will do this everytime you run a command, fetching it in the same request would make the CLI more snappy.

File: packages/eas-cli/src/commands/build/create.ts
Comment (dsokal) [reply packages/eas-cli/src/commands/build/create.ts:48]: @brentvatne what should be the link?

File: packages/eas-cli/src/project/isEasEnabledForProject.ts
Comment (dsokal) [reply packages/eas-cli/src/project/isEasEnabledForProject.ts]: Done

File: packages/eas-cli/src/project/projectUtils.ts
Comment (wkozyra95) [inline packages/eas-cli/src/project/projectUtils.ts:20]: Is it necessary? Do we have cases where this method is called more than once?

In the future, we will probably will add some mechanism of caching responses (mostly for credentials), but I would prefer not to use `once` for that, I think it's better for now to pass this value through some context object and maybe in some cases send request twice.

File: packages/eas-cli/src/project/projectUtils.ts
Comment (wkozyra95) [inline packages/eas-cli/src/project/projectUtils.ts:23]: It's  not specific to this PR, but I think that all functions from this file that call getCOnfig should have this passed as argument.

**Review by wkozyra95 — APPROVED** [review]

File: packages/eas-cli/src/project/projectUtils.ts
Comment (barthap) [reply packages/eas-cli/src/project/projectUtils.ts:20]: Yes, we would have such cases if this PR got merged 😅  For example, in submission command it will be called twice.

File: packages/eas-cli/src/project/projectUtils.ts
Comment (barthap) [reply packages/eas-cli/src/project/projectUtils.ts:23]: We can think about cleaning up `getConfig` calls in whole project - now it's called in many places, during single command execution. I made an issue sometime ago: #81 .
But it's definitely not related to this PR.

**Comment by dsokal** [comment]
@fson I reverted the commit addressing your feedback. Please see the discussion under https://github.com/expo/universe/pull/6477

File: packages/eas-cli/src/commands/build/create.ts
Comment (dsokal) [reply packages/eas-cli/src/commands/build/create.ts:49]: @fson I'll open an issue to add the link after merging this PR.

**Comment by wschurman** [comment]
LGTM but I'll let others who work on the CLI review.

**Review by wkozyra95 — APPROVED** [review]

File: packages/eas-cli/src/project/isEasEnabledForProject.ts
Comment (fson) [inline packages/eas-cli/src/project/isEasEnabledForProject.ts:19]:
```suggestion
  } catch (error) {
    if (error.response?.statusCode === 404) {
      return true;
    } else {
      throw error;
    }
```
It's possible for the `try` block to throw an error that doesn't originate from the HTTP library, so the `?` in `err.response?.statusCode` is needed. Additionally used a full word for the variable name.

File: packages/eas-cli/src/project/projectUtils.ts
Comment (fson) [reply packages/eas-cli/src/project/projectUtils.ts:19]: I think we should not use `once` here, it can cause many kinds of issues.

The most obvious problem is that the arguments of subsequent calls are completely ignored. Example:
```ts
await getProjectIdAsync('/home/ville/super-important-project');

// Later...
const id = await getProjectIdAsync('/home/ville/hobby-project');
await permanentlyDeleteProjectAsync(id); // Deletes super-important-project 😞 
```

Additionally using `once` to cache the result creates a hidden cache inside the function that can be hard to invalidate and can cause surprising results in tests, for example. We've also had some discussions about using the cache of the GraphQL client, which is more widely applicable and allows setting up a normalized cache that will ensure the cached value is updated if a following query returns a new value.

I would suggest just making sure each command only calls this function once and stores the result in a variable passed to other functions.

**Review by fson — CHANGES_REQUESTED** [review]

**Review by fson — APPROVED** [review]

**Review by wkozyra95 — APPROVED** [review]

**Pull request merged**
